client/src/components/ProductList.js

```diff
@@ -12,13 +12,25 @@ export default () => {
       data={products}
       renderItem={({ item }) => (
         <Button onPress={() => navigation.navigate('Product', { id: item._id })}>
-          <Image
+          {item.images[0] ? (
+            <Image
+              source={{
+                uri: item.images[Math.floor(Math.random() * item.images.length)].url,
+              }}
+              alt="product"
+              size="xl"
+            />
+          ) : (
+            ''
+          )}
```

Comment:
It is common to use array length for checking if it is empty or not.
```suggestion
          {item.images.length > 0 && (
            <Image
              source={{
                uri: item.images[Math.floor(Math.random() * item.images.length)].url,
              }}
              alt="product"
              size="xl"
            />
          )}
```